src/IQFeed.CSharpApiClient/Streaming/Level2/Messages/TimestampMessage.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Globalization;
+using IQFeed.CSharpApiClient.Extensions;
+
+namespace IQFeed.CSharpApiClient.Streaming.Level2.Messages
+{
+    // T,[YYYYMMDD HH:mm:SS]<LF>
+    public class TimestampMessage
+    {
+        static char[] chr = { '\r' };
+        public const string TimestampMessageDateTimeFormat = "yyyyMMdd HH:mm:ss";
+
+        public TimestampMessage(DateTime timestamp)
+        {
+            Timestamp = timestamp;
+        }
+
+        public DateTime Timestamp { get; }
+
+        public static TimestampMessage Parse(string message)
+        {
+            var values = message.SplitFeedMessage();
+            
+            DateTime.TryParseExact(values[1].TrimEnd(chr), TimestampMessageDateTimeFormat, CultureInfo.InvariantCulture, DateTimeStyles.None, out var timestamp);
```

Comment:
do you remember the reason why need to strip this character ? looks like IQFeed doesn't follow a certain messaging standard because we don't need this for L1